### ISSUE TITLE
[1.11.2 v1.2.1] - Server crash?

### ISSUE THREAD

**Issue by VallenFrostweaver** [capnkirok/animania#295]
No idea if this even belongs here but I'm getting crashes every couple hours on a server playing Raxpack 2 with Forge 13.20.1.2393

I'm guessing it isn't Animania but it is mentioned in the crash report: https://pastebin.com/d8kjjnai

**Comment by capnkirok** [comment]
Yeah, that's a strange one. Thanks for reporting @VallenFrostweaver and the crash log, will take a closer look.

**Comment by VallenFrostweaver** [comment]
Ok. Strange this. It was due to Mekanism. I've reported it on their github (see below link). I'll close this out.

https://github.com/aidancbrady/Mekanism/issues/4600

**Comment by VallenFrostweaver** [comment]
The Mekanism peeps are saying this is Animania. You can see all the info on it at the above link.

**Comment by capnkirok** [comment]
Thanks for the additional link. At first blush, it does look like Animania. Could be related to #254

**Comment by capnkirok** [comment]
------ Pasted from above thread on Mekanism -------

TL;DR: Weird. Gonna make a quick change, might fix it

Full version:
This is a java.lang.error, which means it doesn't like something we're doing in Java. But on this particular line, the only possibly offensive thing in Java is "equalsIgnoreCase"

if (entity != null && entity.getPersistentID().toString().equalsIgnoreCase(((EntityBullHereford) this.thisAnimal).getMateUniqueId().toString())...

We can just use 'equals' here since we're only comparing UUIDs. I did read something about 'equalsIgnoreCase' being quite slow and also can cause issues in certain locales... so will remove. Hope it works.

**Comment by thiakil** [comment]
@capnkirok why are you even comparing UUIDs with strings? Using UUID#equals would be far more efficient

**Comment by capnkirok** [comment]
Yeah, I am not entirely sure how I got to comparing Strings in the first place (I am sure there was a reason), but indeed, I only use equals now as mentioned.

**Comment by capnkirok** [comment]
This is fixed in 1.3, which is due to release by end of August.
